src/hooks/useNotificationModal.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  SaveSubscriptionMutation,
+  SaveSubscriptionMutationVariables,
+} from "gql/generated/types";
+import { useBannerDispatchContext } from "context/banners";
+import { useState, useEffect } from "react";
+import { useMutation } from "@apollo/react-hooks";
+import { SAVE_SUBSCRIPTION } from "gql/mutations/save-subscription";
+import get from "lodash.get";
+
+export interface UseNotificationModalProps {
```

Comment:
nit. They arent really props more so params since its a hook function and not a react component.